File: python/35e_1399B_GiftsFixing.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = input().split()
        b = input().split()
        min_a, min_b = int(a[0]), int(b[0])
        for i in range(n):
            a[i] = int(a[i])
            b[i] = int(b[i])
            min_a = min(min_a, a[i])
            min_b = min(min_b, b[i])
        cnt = 0
        for i in range(n):
            cnt += max((a[i] - min_a),(b[i] - min_b))
        ans.append(cnt)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/35e_1399B_GiftsFixing.cpp
#include <bits/stdc++.h>

int main() {
    long long temp, min_a, min_b, cnt;
    int t, n;
    std::vector<long long> ans;
    std::cin >> t;

    for (int i = 0; i < t; i++) {
        std::cin >> n;
        std::vector<long long> a, b;

        for (int j = 0; j < n; j++) {
            std::cin >> temp;
            a.push_back(temp);

            if (j == 0) {
                min_a = temp;
            }

            min_a = std::min(min_a, temp);
        }

        for (int j = 0; j < n; j++) {
            std::cin >> temp;
            b.push_back(temp);

            if (j == 0) {
                min_b = temp;
            }

            min_b = std::min(min_b, temp);
        }

        cnt = 0;

        for (int j = 0; j < n; j++) {
            cnt += std::max((a[j] - min_a), (b[j] - min_b));
        }

        ans.push_back(cnt);
    }

    for (auto i : ans) {
        std::cout << i << "\n";
    }

    return 0;
}
